main/Main.hs

```diff
@@ -116,6 +105,24 @@ main = do
   -- Config that can change at runtime
   refConf <- newIORef conf
 
+  -- re-read and override the config with db settings if db-settings-prefix is provided
+  when (isJust dbSettingsPrefix) $
+    reReadConfig pool dbSettingsPrefix env (cliPath opts) refConf
```

Comment:
Didn't see a better choice than to re-read here. The db-uri must be obtained before loading the settings from the db, then the config can be overriden with the db settings.

Reply:
I don't see another way right now either.